### DISCUSSION TITLE
Advanced reverse proxy configuration

### DISCUSSION THREAD

**Discussion by rpspall** [nrwl/nx#14537] [General]
Now I confess I am a newbie to nx and only started with React 6 months ago! I like Nx and really want to stick with it!

Anyway, currently migrating a react application to an Nx integrated workspace, and having an issue with advanced proxy configuration.

I understand that for a simple proxy...

1. Specify the proxy.conf.json file
"serve": {
      "executor": "@nrwl/webpack:dev-server",
      "defaultConfiguration": "development",
      "options": {
        "verbose": true,
        "buildTarget": "ydsca:build",
        "hmr": true,
        **"proxyConfig": "apps/ydsca/proxy.conf.json",**
        "ssl": true,
        "sslKey": "certs/localhost.key.pem",
        "sslCert": "certs/localhost.crt.pem"
      },

2. Implement the proxy.conf.json file, e.g....
{
    "/caconnect/common/*": {
        "target": "https://127.0.0.1:9443",
        "secure": false,
        "logLevel": "debug"
    }
}

This works without any issues, however, the current React application uses "http-proxy-middleware" and a more advanced setupProxy.js file (wildcard matching such as "/**/common/" for any path that contains /common/):

const { createProxyMiddleware } = require('http-proxy-middleware');

module.exports = function(app) {
  app.use(
    [
      '/**/common/',
      '/**/commonClassic/',
      '/**/ydscaCommon/',
      '/ymca',
      '/staff'
    ],
    createProxyMiddleware({
      //Must use 127.0.0.1 as for some reason, localhost stopped working in node versions > 16
      target: 'https://127.0.0.1:9443',
      //Necessary for https with self signed certificates
      secure: false,
      changeOrigin: false,
      //Additional logging stuff!
      onProxyReq: function onProxyReq(proxyReq, req, res) {
          // Log outbound request to remote target
          console.log('-->  ', req.method, req.path, '->', proxyReq.baseUrl + proxyReq.path);
      },
      onError: function onError(err, req, res) {
          console.error(err);
          res.status(500);
          res.json({error: 'Error when connecting to remote server.'});
      },
      //logLevel: 'debug'
    })
  );
 };

How can the same advanced proxying be configured in Nx?

Tried changing proxyConfig": "apps/ydsca/proxy.conf.json" to proxyConfig": "apps/ydsca/setupProxy.js", but it doesn't seem to like javascript files for proxy configuration!

Any suggestions?

Thank you.

**Comment by xiaozhen123321** [comment]
Did you find the answer? I had the same problem
